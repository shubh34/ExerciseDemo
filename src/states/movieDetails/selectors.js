import get from 'lodash/get';
import { createSelector } from 'reselect';

const getMovieDetails = state => get(state, 'movieDetails', '');

export const getHighLightedScore = createSelector(getMovieDetails, movieDetails => get(movieDetails, 'data.scores', []).find(score => score.highlighted));

export const getMovieTitle = createSelector(getMovieDetails, movieDetails => get(movieDetails, 'data.title', ''));

export const getSnapshotImageUrl = createSelector(getMovieDetails, movieDetails => get(movieDetails, 'data.images.snapshot', '').replace('.jpeg', '-width1920-quality80.jpeg'));

export const getMovieDescription = createSelector(getMovieDetails, movieDetails => get(movieDetails, 'data.plot', ''));

export const getScore = createSelector(getHighLightedScore, scores => get(scores, 'score', ''));

export const getVotes = createSelector(getHighLightedScore, scores => get(scores, 'formatted_amount_of_votes', ''));
