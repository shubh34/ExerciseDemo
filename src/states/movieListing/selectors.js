import get from 'lodash/get';

export const getMovieListHeader = (state, id) => get(state, `movieListing[${id}].data.name`);

export const getMovieListByCategory = (state, id) => get(state, `movieListing[${id}].data.contents.data`, []).map(movie => ({
	artwork: movie.images.artwork.replace('.jpeg', '-width217-quality80.jpeg'),
	id: movie.id,
	score: movie.highlighted_score.score,
	votes: movie.highlighted_score.formatted_amount_of_votes,
}));
