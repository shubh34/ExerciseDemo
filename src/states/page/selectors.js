export const getCurrentPage = state => state.page.name;
export const getBackPage = state => state.page.backPage;
