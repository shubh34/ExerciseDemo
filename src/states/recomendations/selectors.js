import get from 'lodash/get';
import { createSelector } from 'reselect';

export const getRecommendedMovies = state => get(state, 'recomendations.data', []);

export const getFormattedRecomendedMovieDetails = createSelector(
	getRecommendedMovies,
	movies => movies.map(movieDetails => ({
		id: movieDetails.id,
		artwork: movieDetails.images.artwork.replace('.jpeg', '-width217-quality80.jpeg'),
		score: movieDetails.highlighted_score.score,
		votes: movieDetails.highlighted_score.formatted_amount_of_votes
	}))
);
