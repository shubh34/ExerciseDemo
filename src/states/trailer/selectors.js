import get from 'lodash/get';

export const getMovieTrailerUrl = state => get(state, 'trailer.trailer.data.stream_infos[0].url');
