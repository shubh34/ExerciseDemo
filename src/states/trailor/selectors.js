import get from 'lodash/get';

export const getMovieTrailorUrl = state => get(state, 'trailor.trailor.data.stream_infos[0].url');
