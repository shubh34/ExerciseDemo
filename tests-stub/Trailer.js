export const trailer = {
	data: {
		type: 'streamings',
		id: 'a67e0a16-b1b9-4a92-845a-bec469899573',
		heartbeat: {},
		stream_infos: [{
			player: 'web:PD-NONE',
			display_aspect_ratio: '1.777777777777778',
			all_subtitles: [],
			cdn: 'VERIZOBS',
			url: 'https://prod-stpeter-pmd.verizon.cdn.rakuten.tv/7/c/3/7c38f0f5b4e1bb200ebb4fa4f6386669-mc-0-137-0-0_SD_TRAILER_PAR_1_1/7c38f0f5b4e1bb200ebb4fa4f6386669-mc-0-137-0-0_SD_TRAILER_PAR_1_1.mp4?UeKaKfcJXUl4EZ34s90Y3c_LOwBF17L76uXbxJmZcPVuu1cmxTXM1YPrR_-tWcASBG7kfDEarl4c5pqAY4a6NrlUPLlVP_mzvcWZ7XfF9a0u7wu7h-dlRysR_VEe_-Q0F9osaJIWYtAREIqGMWbT-Zdq2jOn3ubbvd-xJvcr6NgidJa5w9lYas-7kUdzgEF-O9x0Rv4KadmKPVpRU7qNykIaso8RiQQon4NS5iuPAkh8xGvTGq6xCFKupMEFPhZk2PNr96ThUNRIJg==',
			wrid: 'b436d639-3d4d-43ac-8ab4-abc30bb9762d',
			video_quality: 'SD',
		}],
	},
};
