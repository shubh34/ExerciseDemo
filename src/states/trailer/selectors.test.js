import { getMovieTrailerUrl } from './selectors';
import { trailer } from '../../../tests-stub/Trailer';

describe('Trailer Selector', () => {
	it('should return trailer url', () => {
		const state = { trailer };
		expect(getMovieTrailerUrl(state)).toMatchSnapshot();
	});
});
