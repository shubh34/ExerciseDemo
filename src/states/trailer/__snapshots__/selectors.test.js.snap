// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Trailer Selector should return trailer url 1`] = `"https://prod-stpeter-pmd.verizon.cdn.rakuten.tv/7/c/3/7c38f0f5b4e1bb200ebb4fa4f6386669-mc-0-137-0-0_SD_TRAILER_PAR_1_1/7c38f0f5b4e1bb200ebb4fa4f6386669-mc-0-137-0-0_SD_TRAILER_PAR_1_1.mp4?UeKaKfcJXUl4EZ34s90Y3c_LOwBF17L76uXbxJmZcPVuu1cmxTXM1YPrR_-tWcASBG7kfDEarl4c5pqAY4a6NrlUPLlVP_mzvcWZ7XfF9a0u7wu7h-dlRysR_VEe_-Q0F9osaJIWYtAREIqGMWbT-Zdq2jOn3ubbvd-xJvcr6NgidJa5w9lYas-7kUdzgEF-O9x0Rv4KadmKPVpRU7qNykIaso8RiQQon4NS5iuPAkh8xGvTGq6xCFKupMEFPhZk2PNr96ThUNRIJg=="`;
